include/inviwo/core/datastructures/bitset.h

```diff
@@ -0,0 +1,399 @@
+     */
+    size_t andNotCardinality(const BitSet& b) const;
+    /**
+     * Compute the cardinality of the symmetric differencebetween this bitset and \p b
+     */
+    size_t xorCardinality(const BitSet& b) const;
+
+    /**
+     * Computes the Jaccard index between this bitset and \p b. This distance is also known as the
+     * Tanimoto distance, or the Jaccard similarity coefficient.
+     *
+     * The Jaccard index is undefined if both bitsets are empty.
+     */
+    double jaccardIndex(const BitSet& b) const;
+
+    /**
+     * Intersection of this bitset and \p b
+     */
+    BitSet& operator&=(const BitSet& b);
+    /**
+     * Intersection of this bitset and \p b
+     */
+    BitSet operator&(const BitSet& b) const;
+
+    /**
+     * Difference of this bitset and \p b
+     */
+    BitSet& operator-=(const BitSet& b);
+    /**
+     * Difference of this bitset and \p b
+     */
+    BitSet operator-(const BitSet& b) const;
+
+    /**
+     * Union of this bitset and \p b
+     */
+    BitSet& operator|=(const BitSet& b);
+    /**
+     * Union of this bitset and \p b
+     */
+    BitSet operator|(const BitSet& b) const;
+
+    /**
+     * Symmetric union of this bitset and \p b
+     */
+    BitSet& operator^=(const BitSet& b);
+    /**
+     * Symmetric union of this bitset and \p b
+     */
+    BitSet operator^(const BitSet& b) const;
+
+    /**
+     * Convert bitset to a std::vector holding only set elements
+     */
+    std::vector<uint32_t> toVector() const;
+    /**
+     * Create a string from the bitset
+     */
+    std::string toString() const;
+
+    /**
+     * Return the number of bytes required to serialize the bitset
+     *
+     * \see optimize(), shrinkToFit()
+     */
+    size_t getSizeInBytes() const;
+
+    /**
+     * Write this bitset to stream \p os. The first four byte value contains the size of the bitset
+     * in bytes followed by the binary encoding.
+     */
+    void writeData(std::ostream& os) const;
+
+    /**
+     * Read a BitSet from stream \p is. The first four byte value contains the size of the bitset in
+     * bytes.
+     *
+     * @throw Exception
+     */
+    void readData(std::istream& is);
+
+    friend std::ostream& operator<<(std::ostream& os, const BitSet& b) {
+        os << b.toString();
+        return os;
+    }
+
+    /**
+     * Optimize the bitset representation using run-length encoding based on space efficiency.
+     * Calling shrinkToFit() might result in further memory savings.
+     */
+    void optimize();
+
+    /**
+     * Enforce removing the run-length encoding of the bitset even though it might be more
+     * efficient.
+     *
+     * \see optimize()
+     */
+    void removeRLECompression();
+
+    /**
+     * Reallocates the memory needed.
+     *
+     * @return number of bytes saved
+     */
+    size_t shrinkToFit();
+
+private:
+    void addMany(size_t size, const uint32_t* data);
+
+    struct IVW_CORE_API RoaringDeleter {
```

Comment:
No need to export this private struct?

Reply:
Wrong :-) There is a need for it. Otherwise the linker complains about it when trying to destruct a unique_ptr of Roaring (that one is only forward declared).